packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx

```diff
@@ -79,9 +81,22 @@ export const CloudPulseResourcesSelect = React.memo(
 
     return (
       <Autocomplete
-        onChange={(e, resourceSelections: CloudPulseResources[]) => {
+        onChange={(
+          _: React.SyntheticEvent,
+          resourceSelections: CloudPulseResources[]
+        ) => {
           setSelectedResources(resourceSelections);
-          handleResourcesSelection(resourceSelections, savePreferences);
+
+          if (!isAutocompleteOpen.current) {
```

Comment:
can we put a debounce logic to handleResourceSelection instead of using ref. This handle change seems to be executing after the autocomplete will close which make an additional user click.

Reply:
This can be brainstormed, but this is as per the discussion with kiran.